pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -1486,7 +1486,14 @@ func (r *ReconcileClusterDeployment) mergePullSecrets(cd *hivev1.ClusterDeployme
 	}
 
 	// Check if global pull secret from env as it comes from hive config
-	globalPullSecret := os.Getenv("GLOBAL_PULL_SECRET")
+	globalPullSecretName := os.Getenv(constants.GlobalPullSecret)
```

Comment:
I would **much** rather the global pull secret name came into the controller via dependency injection rather than having the controller read an environment variable.

Reply:
Sorry, did not get your point. Can you please elaborate?

Reply:
An environment variable is logically similar to a command-line argument. It is something that should be read during the startup of the application and then passed around as a parameter where needed. This allows for validation of user inputs in one early place. It also makes it easier to unit test components by reducing the coupling to external resources.

So, rather than having the clusterDeployment controller reading the name of the global pull secret directly from the OS environment, the controller should be created with the name of the global pull secret.

Reply:
Currently the deployment for controllers is getting created at https://github.com/openshift/hive/blob/master/pkg/operator/hive/hive.go#L86 , also check https://github.com/openshift/hive/blob/master/pkg/operator/hive/hive.go#L43. We need to do changes to put the framework which would allow the model you are suggesting.